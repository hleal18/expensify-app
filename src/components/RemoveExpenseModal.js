import React from 'react';
import Modal from 'react-modal';

class RemoveExpenseModal extends React.Component {
    onAcceptActionHandler = () => {
        this.props.onAction();
    };
    render() {
        return (
            <div>
                <Modal
                    isOpen={!!this.props.showModal}
                    onRequestClose={this.props.onCloseModal}
                    contentLabel="Remove Expense"
                    className="modal"
                >
                    <h3 className="modal__title">Do you want to delete the expense?</h3>
                    <button onClick={this.onAcceptActionHandler} className="button">
                        Accept
                    </button>
                </Modal>
            </div>
        );
    }
}

export default RemoveExpenseModal;