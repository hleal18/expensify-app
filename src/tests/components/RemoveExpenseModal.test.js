import React from 'react';
import { shallow } from 'enzyme';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';

let wrapper, onAction, onRequestClose, showModal;

beforeEach(() => {
    onAction = jest.fn();
    onRequestClose = jest.fn();
    showModal = true;
    wrapper = shallow(<RemoveExpenseModal
        onAction={onAction}
        onCloseModal={onRequestClose}
        showModal={showModal}
    />);
});

//Evalua que se renderice correctamente
test('should render RemoveExpenseModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

//Evalua que se accione correctamente la funcion
//onAcceptActionHandler con el evento click del boton
//y en consecuencia, la invocacion de prop.onAction()
test('should test the click event and the invocation of the onAction prop', () => {
    wrapper.find('button').simulate('click');
    expect(onAction).toHaveBeenCalledWith();
});

//Evalua que el modal invoque el mÃ©todo que se encarga
//del cierre del mismo.
test('should invoke the onRquestClose method', () => {
    wrapper.find('Modal').prop('onRequestClose')();
    expect(onRequestClose).toHaveBeenCalledWith();
});