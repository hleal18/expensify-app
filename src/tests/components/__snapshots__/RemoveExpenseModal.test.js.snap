// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal correctly 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={0}
    contentLabel="Remove Expense"
    isOpen={true}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnOverlayClick={true}
  >
    <h3
      className="modal__title"
    >
      Â¿Do you want to delete the expense?
    </h3>
    <button
      className="button"
      onClick={[Function]}
    >
      Accept
    </button>
  </Modal>
</div>
`;
